### ISSUE TITLE
can't edit netdata conf file

### ISSUE THREAD

**Issue by pavs** [firehol/netdata#819]
No matter how many time I edit `/etc/netdata/netdata.conf` file when I run netdata it always loads the default conf instead of loading the edited one. I checked using `<ip>:<port>/netdata.conf`

**Comment by ktsaou** [comment]
I think you are not removing the comment `#` in front of the lines you edit.

**Comment by pavs** [comment]
I did, I have logged my issues in this netdata plugin page.
https://github.com/5310/netdata-plugin-ping/issues/1#issuecomment-241406012

**Comment by pavs** [comment]
I did a fresh install. same issue.

**Comment by ktsaou** [comment]
ok.

I merged PR #820 with some more debugging info added.

pull it again, install it and run this:

```sh
netdata -k -W debug_flags=0xffffffff >/tmp/debug.log 2>&1
```

In that file you can find the filename netdata reads, but also exactly what netdata parses from the config file.

**Comment by ktsaou** [comment]
Just to make sure we do not have a bug somewhere: which option of the config file is not respected?

**Comment by pavs** [comment]
@ktsaou nothing is loaded. It using the default config file that comes with everything commented out. Even when I remove hash for 10+ lines - no change shows on the config file on URL, which includes custom changes I make on plugin options.

**Comment by ktsaou** [comment]
wait a moment. If you just uncomment the default values, they will come back commented again. They are the default values.

You have to change something in the values for them to overwrite the defaults, in which case netdata will respect them and present the new file with these lines uncommented.

**Comment by pavs** [comment]
I also made changes. They didn't show. I will run debug and get back to you in few mins.

**Comment by pavs** [comment]
Ok, this is what I did. 

- git pull
- Ran install script
- Ran debug `sudo /usr/sbin/netdata -k -W debug_flags=0xffffffff >/tmp/debug2.log 2>&1`
- it correctly shows ping plugin being loaded `cat /tmp/debug2.log | grep ping`
```
16-08-22 21:59:30: DEBUG (0150@appconfig.:config_value_cr): netdata: Creating config entry for name 'ping', value 'yes', in section 'plugins'.
16-08-22 21:59:30: DEBUG (0122@appconfig.:config_section_): netdata: Creating section 'plugin:ping'.
16-08-22 21:59:30: DEBUG (0150@appconfig.:config_value_cr): netdata: Creating config entry for name 'update every', value '2', in section 'plugin:ping'.
16-08-22 21:59:30: DEBUG (0150@appconfig.:config_value_cr): netdata: Creating config entry for name 'command options', value 'google.com github.com', in section 'plugin:ping'.
```
- Ran in regular mode `killall netdata` and then `sudo /usr/sbin/netdata` check config through url the cnages doesn't show. grrrrr.
- Ran uninstall script.
- During the delete prompt I type 'y' instead of just pressing enter. Meaning before when I uninstalled the folders and files weren't deleted. Not that it wouldn't make any different because the config wasn't working anyways. But I missed this so I thought I should let you know.
- Changed default port. Added ping plugin. start netdata, it works on new port and changes shows up in config file.
- Even though plugin changes shows up on config file I don't see and ping graph (probably not related to this issue so I am not going to pursue this here). If there is a way I could get in touch with you in privately I could share my url.

You can close this issue, because the problem was solved, even though I don't know what was causing it. Reinstalling again seems to have done the trick.

**Comment by ktsaou** [comment]
Run your ping plugin by hand and give me 1-2 iterations.
In which language is it written?

Does netdata error.log say anything about your plugin?

**Comment by pavs** [comment]
Output from ping plugin:

```
sudo /usr/libexec/netdata/plugins.d/ping.plugin 1 google.com github.com
CHART net.ping '' 'Ping' 'ms' 'ping' '' area
DIMENSION host0 'google.com'
DIMENSION host1 'github.com'
BEGIN net.ping
END
BEGIN net.ping
SET host0 = 81.1
SET host1 = 331
END
BEGIN net.ping
SET host0 = 86.1
SET host1 = 332
END
BEGIN net.ping
SET host0 = 81.9
SET host1 = 332
```

This is the plugin I am using: https://github.com/5310/netdata-plugin-ping

Forgot to check error log. There is something:

```
/usr/lib/node_modules/netdata-plugin-ping/ping.plugin:4
16-08-22 22:47:48: INFO: netdata: PLUGINSD: '/usr/libexec/netdata/plugins.d/ping.plugin' on pid 27244 stopped after 0 successful data collections (ENDs).
16-08-22 22:47:48: ERROR: netdata: PLUGINSD: '/usr/libexec/netdata/plugins.d/ping.plugin' exited with error code 8. Disabling it.
16-08-22 22:47:48: python.d ERROR: local already exists. Stopping 'localhost'
```

**Comment by ktsaou** [comment]
Check this: http://stackoverflow.com/questions/21960111/node-js-express-npm-start-error-exit-status-8

**Comment by ktsaou** [comment]
You can try to run your plugin as user netdata:

```
su netdata -s /bin/bash
```

then run the plugin in the new prompt.

**Comment by pavs** [comment]
@ktsaou Is this an ownership issue? I tried changing ping.plugin ownership to `netdata:netdata` but doesn't seems to fix it. Anything else I can do?

```
/usr/libexec/netdata/plugins.d$ ls -lh
total 196K
-rwxr-xr-x 1 root    root     10K Aug 23 12:56 alarm-email.sh
-rwxr-xr-x 1 root    root    107K Aug 23 12:56 apps.plugin
-rwxr-xr-x 1 root    root    2.4K Aug 23 12:56 cgroup-name.sh
-rwxr-xr-x 1 root    root    1.2K Aug 23 12:56 charts.d.dryrun-helper.sh
-rwxr-xr-x 1 root    root     20K Aug 23 12:56 charts.d.plugin
-rwxr-xr-x 1 root    root    2.4K Aug 23 12:56 loopsleepms.sh.inc
-rwxr-xr-x 1 root    root     11K Aug 23 12:56 node.d.plugin
lrwxrwxrwx 1 netdata netdata   59 Aug 23 13:42 ping.plugin -> /usr/local/lib/node_modules/netdata-plugin-ping/ping.plugin
-rwxr-xr-x 1 root    root     19K Aug 23 12:56 python.d.plugin
-rw-r--r-- 1 root    root    7.3K Aug 23 12:56 README.md
-rwxr-xr-x 1 root    root    2.8K Aug 23 12:56 tc-qos-helper.sh
```

```
 ls -lh /usr/local/lib/node_modules/netdata-plugin-ping/
total 20K
-rwxr-xr-x 1 pavs    pavs    1.1K Apr  6 16:54 license
drwxrwxr-x 3 pavs    pavs    4.0K Aug 23 13:34 node_modules
-rwxr-xr-x 1 pavs    pavs    1.9K Aug 23 13:34 package.json
-rwxr-xr-x 1 netdata netdata 1.5K Apr  6 16:54 ping.plugin
-rwxr-xr-x 1 pavs    pavs    1008 Apr  6 16:54 readme.md
```

**Comment by pavs** [comment]
When trying to run plug.ping as user netdata

```
su netdata -s /bin/bash
netdata@octopus:/usr/libexec/netdata/plugins.d$ /usr/libexec/netdata/plugins.d/ping.plugin

/usr/local/lib/node_modules/netdata-plugin-ping/ping.plugin:4
const rx = require('rx').Observable
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
netdata@octopus:/usr/libexec/netdata/plugins.d$
```

**Comment by jacksonn55** [comment]
This guy has been hacking in my phone and I don't know how he's doing it. I
have the same phone he does but I have a newer version.

**Comment by pavs** [comment]
wat?

**Comment by pavs** [comment]
FIXED IT! Tried way too many things. Long story short I had to reinstall nodejs and change file permission for rx module for node. I don't know enough if this the right way of doing things. I am just happy that it finally worked. :)


![screen shot 2016-08-23 at 10 26 12 pm](https://cloud.githubusercontent.com/assets/3807064/17900320/adab9576-6980-11e6-8077-e719c13ce2b5.png)

Thanks again for being patient with me. Really appreciate it and appreciate all the wonderful work you guys are doing with netdata. Keep it up.
